### PR TITLE
Add FFSW Support

### PR DESCRIPTION
**Pull request by owainlewis**

### CODE REVIEW

File: pkg/provisioner/filestorage/filestorage.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeReclaimPolicy: options.PersistentVolumeReclaimPolicy,
+			AccessModes:                   options.PVC.Spec.AccessModes,
+			// FIXME?????
+			Capacity: v1.ResourceList{
+				v1.ResourceName(v1.ResourceStorage): options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)],
+			},
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				FlexVolume: &v1.FlexVolumeSource{
+					Driver: "oracle/oci",
```
Comment (prydie) [inline pkg/provisioner/filestorage/filestorage.go]: Maybe add a constant for this as it's used in both?

File: pkg/provisioner/filestorage/filestorage.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+package filestorage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
```
Comment (prydie) [inline pkg/provisioner/filestorage/filestorage.go]: nit: import grouping

File: pkg/provisioner/filestorage/filestorage.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+// filesystemProvisioner is the internal provisioner for OCI filesystem volumes
+type filesystemProvisioner struct {
+	client client.ProvisionerClient
+}
+
+var _ plugin.ProvisionerPlugin = &filesystemProvisioner{}
+
+// NewFilesystemProvisioner creates a new file system provisioner that creates
+// filsystems using OCI file system service
```
Comment (prydie) [inline pkg/provisioner/filestorage/filestorage.go]: nit: missing `.` (re `go doc`)

File: pkg/provisioner/filestorage/filestorage.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+			Name: mapVolumeIDToName(*response.FileSystem.Id),
+			Annotations: map[string]string{
+				ociVolumeID: *response.FileSystem.Id,
+			},
+			Labels: map[string]string{},
+		},
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeReclaimPolicy: options.PersistentVolumeReclaimPolicy,
+			AccessModes:                   options.PVC.Spec.AccessModes,
+			// FIXME?????
```
[the remark this answers is not in the document]
Comment (garthy) [reply pkg/provisioner/filestorage/filestorage.go]: So the filesystem storage stuff doesn't have a concept of capacity.. So I'm just setting what they passed in but it's a lie!

[the remark this answers is not in the document]
Comment (prydie) [reply pkg/provisioner/filestorage/filestorage.go]: 👍. The [CephFS implementation](https://github.com/oracle/oci-volume-provisioner/blob/e00fe33ed90476d0c48b1ff8312472ca59a6f930/vendor/github.com/kubernetes-incubator/external-storage/ceph/cephfs/cephfs-provisioner.go#L146) seems to do the same thing.

Maybe expand the comment to explain for posterity down the line?

**Pull request merged**